plugin/acl/setup.go

```diff
@@ -0,0 +1,188 @@
+package acl
+
+import (
+	"net"
+	"strings"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/metrics"
+
+	"github.com/caddyserver/caddy"
+	"github.com/infobloxopen/go-trees/iptree"
+	"github.com/miekg/dns"
+)
+
+const (
+	// QtypeAll is used to match any kinds of DNS records type.
+	// NOTE: The value of QtypeAll should be different with other QTYPEs defined in miekg/dns.
+	QtypeAll uint16 = 299
+)
+
+var (
+	// DefaultFilter is the default ip filter which matches both IPv4All and IPv6All.
+	DefaultFilter *iptree.Tree
+)
+
+func init() {
+	initDefaultFilter()
```

Comment:
That we're abusing init for RegisterPlugin is bad enough; In Go it makes sense to minimize this use (of `init`)

Reply:
Will move `initDefaultFilter` out of `init()`.